delocate/libsana.py

```diff
@@ -40,7 +41,11 @@ class DependencyNotFound(Exception):
     """
 
 
-def _filter_system_libs(libname: Text) -> bool:
+def filter_system_libs(libname: Text) -> bool:
```

Comment:
This was duplicated between here and `delocating.py`, so I removed the latter in favor of this copy.

Reply:
Moving `filter_system_libs` like this technically breaks the API. Were you using it in your own library?

Reply:
Oh, that's a good point. Given the circular import issues between these to files, I guess I can just revert back to the duplication.

Reply:
This function is really showing its age by accepting a `Text` input. I suspect it actually works with POSIX paths exclusively. Did something happen to cause this refactor?

I've very hesitant to accept a PR with `os.path.splitdrive`.

Reply:
All of these changes are the result of repairing macOS wheels on a Windows host.

If I remember correctly, callers of `filter_system_libs` are inquiring about, for example, `C:\usr\lib\libSystem.B.dylib` - which obviously doesn't exist on the repairing host - but we can still consider it as a system library. So maybe the proper fix is elsewhere upstream.

Reply:
I'm very interested in learning what's putting `C:` into `C:\usr\lib\libSystem.B.dylib`.

Reply:
I can try to reproduce the issue again tomorrow. Unfortunately I don't actually have a Windows machine so it's a lot of pushing and waiting for CI. :(

Reply:
Well, since you've reopened this:
> I'm very interested in learning what's putting C: into C:\usr\lib\libSystem.B.dylib.

It comes from [here](https://github.com/matthew-brett/delocate/blob/eaba43dc531d0af348cb9b949a6c3f0790d207b2/delocate/libsana.py#L642). On windows, `realpath("foo") == "C:\\foo"`, where `C:\` is the drive letter or UNC path of the current working directory. So if you chdir into `\\myserver\shared_dir`, then `realpath("foo") == "\\\\myserver\\shared_dir\\foo"`

Reply:
Good work. I think `realpath` should not be called on non-existing files in this case.

If I changed the return types for `search_environment_for_lib` then I'd use `Path` for existing files and `PurePosixPath` for missing files.